### PR TITLE
Bypass annoying chrome pop-ups

### PR DESCRIPTION
**Pull request by emtuls**
This fixes https://github.com/mandiant/VM-Packages/issues/977

### CODE REVIEW

**Comment by emtuls** [comment]
This is a bit frustrating that it is erroring out, but I think I finally found the issue.

It looks like the `Windows-2019` and `Windows-2022` both have up to date Chrome installs, and because "chrome is already installed", we are receiving an MSI installer error. Not really sure how to mitigate this...
https://github.com/actions/runner-images/blob/main/images/windows/Windows2019-Readme.md#browsers-and-drivers
https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md#browsers-and-drivers

Might just need to accept that it will fail the install check, but if you test it locally, it should work.

**Comment by emtuls** [comment]
Should we add this to `exclude_tests` in `test_install.ps1` as this is similar to how we are having issues with `windows-terminal`? 

From https://github.com/mandiant/VM-Packages/issues/929, we were looking to exclude it from `daily.yml`, but by putting it into an exclusion inside of `test_install.ps1`, and this file already has an `exclude_tests` that we can use?

**Comment by Ana06** [comment]
This is a nice idea @emtuls!

> It looks like the Windows-2019 and Windows-2022 both have up to date Chrome installs, and because "chrome is already installed", we are receiving an MSI installer error.

I think the error is already in the test suite, see today's daily: https://github.com/mandiant/VM-Packages/actions/runs/8625498921/job/23642184219#step:3:1758

**Comment by Ana06** [comment]
I am not able to reproduce the error locally. do you @emtuls? In any case, I think instead of ignoring testing, we should ignore the error and ensure that the Chrome executable is present.

**Comment by emtuls** [comment]
Locally it shouldn't have an issue, since it is not already installed on your system, whereas it is already installed on the `windows-2019/2022` images that github actions run on.

I see. What would be the best way to ignore the error in the package when it installs? Trying not to change `$ErrorActionPreference = 'Stop'` to just be `Continue` instead, as I don't think that's the best route. 🤔

**Comment by Ana06** [comment]
@emtuls 
> Locally it shouldn't have an issue, since it is not already installed on your system,

What I meant is that I tried to install google chrome in several ways locally and then installing the packages to try to replicate the GH setup and I was not able to reproduce the error.

I think using `-ErrorAction SilentlyContinue` and checking if chrome is installed afterwards is the way to go.

File: packages/googlechrome.vm/tools/chocolateyinstall.ps1 [hunk 1]
```diff
@@ -20,7 +20,33 @@ try {
         fileType        = 'MSI'
         silentArgs      = "/quiet /norestart /l*v `"$($env:TEMP)\$($env:chocolateyPackageName).$($env:chocolateyPackageVersion).MsiInstall.log`""
     }
-    Install-ChocolateyInstallPackage @packageArgs
+    Install-ChocolateyInstallPackage @packageArgs -ErrorAction SilentlyContinue
+
+    $chromePath = "${env:ProgramFiles}\Google\Chrome\Application\chrome.exe"
+    VM-Assert-Path $chromePath
```
Comment (Ana06) [inline packages/googlechrome.vm/tools/chocolateyinstall.ps1:26]: [nitpick] to make it more compact:
```suggestion
    Install-ChocolateyInstallPackage @packageArgs -ErrorAction SilentlyContinue
    VM-Assert-Path "%ProgramFiles%\Google\Chrome\Application\chrome.exe"
```

File: packages/googlechrome.vm/tools/chocolateyinstall.ps1 [hunk 2]
```diff
@@ -20,7 +20,33 @@ try {
...
+    # Create the "Default" folder with a "Preferences" file with data required to bypass "Enhanced ad privacy" pop-up.
+    New-Item -Path $userDataPath -Name "Default" -ItemType Directory -Force -ea 0 | Out-Null
+    $contentOptions = @{
+        Path = Join-Path $userDataPath "Default\Preferences"
+        Value = "`{`"privacy_sandbox`":{`"m1`":{`"row_notice_acknowledged`":true}}`}"
+    }
+    Set-Content @contentOptions
 } catch {
     VM-Write-Log-Exception $_
 }
```
Comment (Ana06) [inline packages/googlechrome.vm/tools/chocolateyinstall.ps1:52]: I think we shouldn't fail the package if setting the configuration fails:

```suggestion
} catch {
    VM-Write-Log-Exception $_
}

# Try to set configuration, but do not fail the package if it fails
$ErrorActionPreference = 'Continue'

# Expand the path to the Chrome User Data folder and create the "User Data" folder if it doesn't exist.
$userDataPath = ${Env:LOCALAPPDATA} + "\Google\Chrome\User Data"
New-Item -ItemType Directory -Path $userDataPath -Force -ea 0 | Out-Null

# Create the empty file "First Run" to bypass the "Sign in to Chrome" initial pop-up.
New-Item -Path $userDataPath -Name "First Run" -ItemType File -Force -ea 0 | Out-Null

# Create the "Local State" file with data required to bypass the "What's New" pop-up tab.
# Number must be higher than the current Chrome Version number and using 999 gives us some time before it fails.
$contentOptions = @{
    Path = Join-Path $userDataPath "Local State"
    Value = '{"browser":{"last_whats_new_version":999}}'
}
Set-Content @contentOptions

# Create the "Default" folder with a "Preferences" file with data required to bypass "Enhanced ad privacy" pop-up.
New-Item -Path $userDataPath -Name "Default" -ItemType Directory -Force -ea 0 | Out-Null
$contentOptions = @{
    Path = Join-Path $userDataPath "Default\Preferences"
    Value = "`{`"privacy_sandbox`":{`"m1`":{`"row_notice_acknowledged`":true}}`}"
}
Set-Content @contentOptions
```

**Comment by emtuls** [comment]
> @emtuls
> 
> > Locally it shouldn't have an issue, since it is not already installed on your system,
> 
> What I meant is that I tried to install google chrome in several ways locally and then installing the packages to try to replicate the GH setup and I was not able to reproduce the error.
> 
> I think using `-ErrorAction SilentlyContinue` and checking if chrome is installed afterwards is the way to go.

Interesting...That was the conclusion I arrived at based on searching the error on the internet, though it could be many things, but it was something that someone mentioned, and I did notice that the Github Action VM's already had it installed. 😬

I'd have to play around with it some more to really try to narrow it down better. I wish I could access the Github Action VM to test from there.

File: packages/googlechrome.vm/tools/chocolateyinstall.ps1 (hunk 1 above)
Comment (emtuls) [reply packages/googlechrome.vm/tools/chocolateyinstall.ps1:26]: Interestingly enough, that seems to fail?
image

**Comment by Ana06** [comment]
@emtuls

> I wish I could access the Github Action VM to test from there.

You can debug GH actions in several ways. I like to run them in my own fork with [logging enabled](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging). Another alternatives are to use the [debugging with ssh action](https://github.com/marketplace/actions/debugging-with-ssh) or [Act](https://github.com/nektos/act).

**Review by Ana06 — APPROVED** [review]
Thanks @emtuls!

**Pull request merged**